### PR TITLE
[Profiler] add views in summary API

### PR DESCRIPTION
**Pull request by kuizhiqing**
### PR types

New features

### PR changes

APIs

### Describe

* add enum type SummaryView
* add `views` in API `summary` to define which SummaryView to show in summary API

Usage
```
from paddle.profiler import SummaryView
...
p.summary(sorted_by=profiler.SortedKeys.GPUTotal, views=[SummaryView.MemoryView, SummaryView.KernelView])
```
Only views in the list will be shown.

**NOTE: the default value of `views`  is None, which make this change compatible.**

### CODE REVIEW

**Review by rainyfly — DISMISSED** [review]
LGTM

File: python/paddle/profiler/profiler.py [hunk 1]
```diff
@@ -734,6 +749,7 @@ def summary(self,
                 sorted_by=SortedKeys.CPUTotal,
                 op_detail=True,
                 thread_sep=False,
+                views=None,
```
Comment (jeff41404) [inline python/paddle/profiler/profiler.py]: In order to ensure compatibility, the new parameter is usually added at the end of API parameters  , but the `views` is not. do we have another considerations?

File: python/paddle/profiler/profiler_statistic.py [hunk 2]
```diff
@@ -698,9 +698,12 @@ def _build_table(statistic_data,
                  sorted_by=SortedKeys.CPUTotal,
                  op_detail=True,
                  thread_sep=False,
+                 views=None,
```
Comment (jeff41404) [inline python/paddle/profiler/profiler_statistic.py]: same problem as above. 
But if this function is not public API and just used by internal calls, that's OK

**Review by jeff41404 — APPROVED** [review]
LGTM

**Review by Ligoml — APPROVED** [review]
LGTM for docs

**Review by sneaxiy — APPROVED** [review]
LGTM

**Pull request merged**